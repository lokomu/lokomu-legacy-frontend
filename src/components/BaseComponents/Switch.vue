<template>
  <div class="flex relative w-full h-full">
    <div
      class="absolute transition-transform duration-1000 ease-in-out rounded-3xl h-full w-1/2"
      :class="
        changed
          ? 'translate-x-full ' + secondClass
          : 'translate-x-0 ' + firstClass
      "
    />
    <input
      id="first"
      type="radio"
      name="option"
      class="peer hidden peer/first"
      :checked="!changed"
    >
    <label
      for="first"
      class="cursor-pointer p-2 text-center z-30 w-1/2 left-0 rounded-full font-bold peer-checked:font-normal"
      :class="firstTextStyle"
      @click="change(false)"
    >{{ firstText }}</label>
    <input
      id="second"
      type="radio"
      name="option"
      class="peer hidden peer/second"
      :checked="changed"
    >
    <label
      for="second"
      class="cursor-pointer p-2 text-center z-30 w-1/2 right-0 rounded-full peer-checked/first:font-bold"
      :class="secondTextStyle"
      @click="change(true)"
    >{{ secondText }}</label>
  </div>
</template>

<script setup lang="ts">
import { ref, watch } from "vue";

const props = defineProps<{
  firstText: string;
  secondText: string;
  firstTextStyle: string;
  secondTextStyle: string;
  firstClass: string;
  secondClass: string;
  rightSideOpen: boolean;
}>();

const changed = ref<boolean>(false);

watch(
  () => props.rightSideOpen,
  (newVal) => {
    changed.value = newVal;
  },
);

const emit = defineEmits<{
  switch: [activate: boolean];
}>();

function change(newValue: boolean): void {
  if (changed.value === newValue) return;
  changed.value = newValue;
  emit("switch", newValue);
}
</script>
