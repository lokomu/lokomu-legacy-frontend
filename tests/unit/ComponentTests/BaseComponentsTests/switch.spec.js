import { shallowMount } from "@vue/test-utils";
import Switch from "@/components/BaseComponents/Switch.vue";
import { describe, expect, it} from "vitest";

describe("AddSwitch component", () => {
  let wrapper = shallowMount(Switch);

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
