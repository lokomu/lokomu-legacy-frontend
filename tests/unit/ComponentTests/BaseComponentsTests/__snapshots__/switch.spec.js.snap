// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddSwitch component > renders correctly 1`] = `
<div
  class="flex relative w-full h-full"
>
  <div
    class="absolute transition-transform duration-1000 ease-in-out rounded-3xl h-full w-1/2 translate-x-0 undefined"
  />
  <input
    class="peer hidden peer/first"
    id="first"
    name="option"
    type="radio"
  />
  <label
    class="cursor-pointer p-2 text-center z-30 w-1/2 left-0 rounded-full font-bold peer-checked:font-normal"
    for="first"
  />
  <input
    class="peer hidden peer/second"
    id="second"
    name="option"
    type="radio"
  />
  <label
    class="cursor-pointer p-2 text-center z-30 w-1/2 right-0 rounded-full peer-checked/first:font-bold"
    for="second"
  />
</div>
`;
